Hierarchical3DRegistration/Hierarchical3DRegistration.py

```diff
@@ -172,78 +207,196 @@ def initializeParameterNode(self) -> None:
+            self.updateRegistrationButtonsState()
+
+            # TODO: Remove this once this is working with the parameterNodeWrapper
+            #  see Slicer issue: https://github.com/Slicer/Slicer/issues/7905
+            currentRootIDStatus = self.ui.SubjectHierarchyComboBox.currentItem()
+            if currentRootIDStatus == 0:
+                raise ValueError("Invalid hierarchy object selected!")
+            self._parameterNode.hierarchyRootID = currentRootIDStatus
+
+            if self._parameterNode.sourceVolume.GetID() == self._parameterNode.volumeSequence.GetID():
+                raise ValueError(f"The source volume must not be the same as the input sequence selected!")
+
+            self.rootBone = TreeNode(
+                hierarchyID=self._parameterNode.hierarchyRootID,
+                ctSequence=self._parameterNode.volumeSequence,
+                sourceVolume=self._parameterNode.sourceVolume,
+                isRoot=True,
+            )
+            self._parameterNode.currentFrame = self._parameterNode.startFrame
+            self.bonesToTrack = [self.rootBone]
+
+            self._parameterNode.runSatus = Hierarchical3DRegistrationRunStatus.IN_PROGRESS
+            self.updateRegistrationButtonsState()
+
+            if self._parameterNode.skipManualTfmAdjustments:
+                self.onRegisterButton()
+            else:
+                nextBone = self.bonesToTrack[0]
+                nextBone.startInteraction(self._parameterNode.currentFrame)
+                self._parameterNode.statusMsg = "Adjust the initial guess transform for the bone " \
+                                                    f"'{nextBone.name}' in frame {self._parameterNode.currentFrame}"
+                slicer.app.processEvents()
+
+    def onRegisterButton(self):
+        with slicer.util.tryWithErrorDisplay("Failed to compute results.", waitCursor=True):
+            currentRootIDStatus = self.ui.SubjectHierarchyComboBox.currentItem() != 0
+            # TODO: Remove this once this is working with the parameterNodeWrapper.
+            #  It's currently commented out due to bug with parameter node, see
+            #  Slice issue: https://github.com/Slicer/Slicer/issues/7905
+            if not currentRootIDStatus:
+                raise ValueError("Invalid hierarchy object selected!")
+            if self._parameterNode.runSatus == Hierarchical3DRegistrationRunStatus.CANCELING:
+                raise ValueError("Canceling registration...")
+
+            self.currentBone = self.bonesToTrack.pop(0)
+            self._parameterNode.currentBoneID = -1  # TOOD: want this meaningfully saved to scene and re-imported later
+
+            self.currentBone.stopInteraction(self._parameterNode.currentFrame)
 
-                    CT = self.ui.inputSelectorCT.currentNode()
-                    rootID = self.ui.SubjectHierarchyComboBox.currentItem()
+            source_frame = self._parameterNode.currentFrame - 1
```

Comment:
source volume is scalar... why frame index?

Reply:
This variable holds the frame index, yes. The code as currently written needs better organization and some cleaning up for clarity.